src/datatypes/interval.coffee

```diff
@@ -165,6 +165,11 @@ module.exports.Interval = class Interval
     else # ol is null
       null
 
+  sameAs: (other, precision) ->
+    if (@ == null or other == null) then return null
+    if (@low == null or @high == null or other.low == null or other.high == null) then return null
```

Comment:
I don't think returning `null` if the low or high is `null` is quite right.  I think it's more complicated than that. According to the spec:
> **Same As**: The same-precision-as operator for intervals returns true if the two intervals start and end at the same value, using the semantics described in the `Start` and `End` operators to determine interval boundaries...

And if you look at the definitions of `Start` and `End`:

> **Start**: ... If the low boundary of the interval is closed and the low value of the interval is not null, this operator returns the low value of the interval. Otherwise, the result is the minimum value of the point type of the interval.

> **End**: ... If the high boundary of the interval is closed and the high value of the interval is not null, this operator returns the high value of the interval. Otherwise, the result is the maximum value of the point type of the interval.

So... in other words, if an interval boundary is _closed_ and its value is _null_, then it's treated like one of the min/max values (because its intentional is really negative or positive infinity).  In this respect, `Interval[@2018-01-01, null] same day as Interval[@2018-01-01, null]` should be `true`.

So... I think you really want to be looking at the calculated `Start` and `End` of the intervals, as the spec suggests.

Reply:
I'm noticing now that it looks like maybe our `Start` and `End` implementations in `interval.coffee` are not right either, as they don't take into account whether the interval boundaries are open or closed.  We should probably fix that (although maybe that's a different pull request).

Looking at the code, I see that the `Interval` class has a `toClosed()` function on it.  It looks like that might actually take care of converting boundaries correctly and giving you an interval that is easier to compare on the low and high values.

Using `toClosed` on both intervals before comparing would also take care of converting open to closed boundaries, which would likely fix another use case that I think is currently broken.  In short, `Interval[@2018-01-01T00:00:00.0, @2019-01-01T00:00:00.0) same as Interval[@2018-01-01T00:00:00.0, @2018-12-31T23:59:59.999]` should also be `true` since the end boundaries are just two different ways of saying the same thing.